dart/test/scope_test.dart

```diff
@@ -230,12 +230,90 @@ void main() {
     expect(sut.extra, clone.extra);
     expect(sut.tags, clone.tags);
     expect(sut.breadcrumbs, clone.breadcrumbs);
+    expect(sut.contexts, clone.contexts);
     expect(ListEquality().equals(sut.fingerprint, clone.fingerprint), true);
     expect(
       ListEquality().equals(sut.eventProcessors, clone.eventProcessors),
       true,
     );
   });
+
+  group('Scope apply', () {
+    final scopeUser = User(
+      id: '800',
+      username: 'first-user',
+      email: 'first@user.lan',
+      ipAddress: '127.0.0.1',
+      extras: <String, String>{'first-sign-in': '2020-01-01'},
+    );
+
+    final breadcrumb = Breadcrumb(message: 'Authenticated');
+
+    setUp(() {});
+
+    test('apply context to event', () {
+      final event =
+          SentryEvent(tags: {'etag': '987'}, extra: {'e-infos': 'abc'});
+      final scope = Scope(SentryOptions())
+        ..user = scopeUser
+        ..fingerprint = ['example-dart']
+        ..addBreadcrumb(breadcrumb)
+        ..transaction = '/example/app'
+        ..level = SentryLevel.warning
+        ..setTag('build', '579')
+        ..setExtra('company-name', 'Dart Inc')
+        ..setContexts('theme', 'material')
+        ..addEventProcessor(
+          (event, hint) => event..tags.addAll({'page-locale': 'en-us'}),
+        );
+
+      final updatedEvent = scope.applyToEvent(event, null);
+
+      expect(updatedEvent.user, scopeUser);
+      expect(updatedEvent.transaction, '/example/app');
+      expect(updatedEvent.fingerprint, ['example-dart']);
+      expect(updatedEvent.breadcrumbs, [breadcrumb]);
+      expect(updatedEvent.level, SentryLevel.warning);
+      expect(updatedEvent.tags,
+          {'etag': '987', 'build': '579', 'page-locale': 'en-us'});
+      expect(
+          updatedEvent.extra, {'e-infos': 'abc', 'company-name': 'Dart Inc'});
+      expect(updatedEvent.contexts['theme'], 'material');
+    });
+
+    test('should not apply context properties when event has already it ', () {
```

Comment:
```suggestion
    test('should not apply context properties when event already has it ', () {
```

Reply:
I'd make this test smaller and more focused on the context field, right now all the user, breadcrumb etc is just noisy for the name of the test

Reply:
I dont see this test testing the `contexts` field though, its not even called

Reply:
✅